### DISCUSSION TITLE
国外VPS该如何解除限制？

### DISCUSSION THREAD

**Discussion by sktkyo** [1715173329/UnblockNeteaseMusic#1511] [General]
我部署在国外的VPS上，国内的机器可以连接上，但是不能解析，不仅VIP不能播放，连普通的歌曲也不行，所以想请教一下各位，作者在readme里面提到的x-real-ip要怎么用？要怎么修改源代码？

**Comment by Limburgish6747** [comment]
你解决了没？我也是在国外vps部署的，我的能正常使用，你照抄我的docker compose配置试试：
services:
  unblock:
    image: pan93412/unblock-netease-music-enhanced:latest
    container_name: unblock
    command: "-o pyncmd kugou kuwo bilibili migu -p 80:443 -a 172.168.10.163 -f 103.135.240.77"
    ports:
      - "8080:80"
      - "8081:443"
    networks:
      music_network:
        ipv4_address: 172.168.10.163
    dns:
      - 8.8.8.8
      - 1.1.1.1
    environment:
      - ENABLE_LOCAL_VIP=true
      - NO_CACHE=true
      - ENABLE_FLAC=true
      - NODE_ENV=production
    deploy:
      resources:
        limits:
          memory: 300M
    logging:
      options:
        max-size: "10m"
    restart: always

networks:
  music_network:
    ipam:
      config:
        - subnet: 172.168.10.0/24
        
我用"networks:"和"-a 172.168.10.163"指定了ip，有其他用途，你可以删掉；另外这个项目有内存溢出，所以我限制了300m内存
直接docker compose up就能使用
root@:/home/unm# docker compose up
[+] Running 1/1
 ✔ Container unblock  Created                                                                                                                          0.0s 
Attaching to unblock
unblock  | INFO: (app) HTTP Server running @ http://172.168.10.163:80
unblock  | INFO: (app) HTTPS Server running @ http://172.168.10.163:443
unblock  | ERROR: (provider/match) 
unblock  |     AggregateError [ETIMEDOUT]: 
unblock  |         at internalConnectMultiple (node:net:1128:18)
unblock  |         at internalConnectMultiple (node:net:1196:5)
unblock  |         at Timeout.internalConnectMultipleTimeout (node:net:1720:5)
unblock  |         at listOnTimeout (node:internal/timers:596:11)
unblock  |         at process.processTimers (node:internal/timers:529:7)
unblock  |     code: ETIMEDOUT
unblock  | INFO: (provider/match) Replaced: [1472480890] 群青
unblock  |     audioId: 1472480890
unblock  |     songName: "群青"
unblock  |     url: "https://m********************************2111b5debe72f0392c2b8fd3.flac"
从日志上来看也没有问题，直接在你的网易云客户端填上代理ip:8080就能使用，要注意的是windows网易云客户端要用2.9.9版本的，不能更新3.0以上，否则用不了

**Reply by Limburgish6747** [reply]
docker compose的缩进有问题，我编辑的时候还是正常的，你复制到gpt里让他帮你排版一下就行

**Comment by sktkyo** [comment]
哦，好的，谢谢，我现在改用BCM了，改天等VPS降价了再去弄一个试试。




雪亲侯
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "UnblockNeteaseMusic/server"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年2月2日(星期天) 晚上9:08
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [UnblockNeteaseMusic/server] 国外VPS该如何解除限制？ (Discussion #1511)





 
你解决了没？我也是在国外vps部署的，我的能正常使用，你照抄我的docker compose配置试试：
 services:
 unblock:
 image: pan93412/unblock-netease-music-enhanced:latest
 container_name: unblock
 command: "-o pyncmd yt-dlp kugou kuwo bilibili migu -p 80:443 -a 172.168.10.163 -f 103.135.240.77"
 ports:
 - "8080:80"
 - "8081:443"
 networks:
 music_network:
 ipv4_address: 172.168.10.163
 dns:
 - 8.8.8.8
 - 1.1.1.1
 environment:
 - ENABLE_LOCAL_VIP=true
 - NO_CACHE=true
 - ENABLE_FLAC=true
 - NODE_ENV=production
 deploy:
 resources:
 limits:
 memory: 300M
 logging:
 options:
 max-size: "10m"
 
networks:
 music_network:
 ipam:
 config:
 - subnet: 172.168.10.0/24
 
我用"networks:"和"-a 172.168.10.163"指定了ip，有其他用途，你可以删掉；另外这个项目有内存溢出，所以我限制了300m内存
 直接docker compose up就能使用
 root@:/home/unm# docker compose up
 [+] Running 1/1
 ✔ Container unblock  Created                                                                                                                          0.0s
 Attaching to unblock
 unblock  | INFO: (app) HTTP Server running @ http://172.168.10.163:80
 unblock  | INFO: (app) HTTPS Server running @ http://172.168.10.163:443
 unblock  | ERROR: (provider/match)
 unblock  |     AggregateError [ETIMEDOUT]:
 unblock  |         at internalConnectMultiple (node:net:1128:18)
 unblock  |         at internalConnectMultiple (node:net:1196:5)
 unblock  |         at Timeout.internalConnectMultipleTimeout (node:net:1720:5)
 unblock  |         at listOnTimeout (node:internal/timers:596:11)
 unblock  |         at process.processTimers (node:internal/timers:529:7)
 unblock  |     code: ETIMEDOUT
 unblock  | INFO: (provider/match) Replaced: [1472480890] 群青
 unblock  |     audioId: 1472480890
 unblock  |     songName: "群青"
 unblock  |     url: "https://m701.music.126.net/20250202212708/64ed419482dc3ce9a296cf270d2980e9/jdymusic/obj/wo3DlMOGwrbDjj7DisKw/22259854913/a85d/05ad/5213/3a06b3082111b5debe72f0392c2b8fd3.flac"
 从日志上来看也没有问题，直接在你的网易云客户端填上代理ip:8080就能使用，要注意的是windows网易云客户端要用2.9.9版本的，不能更新3.0以上，否则用不了
